tests/unit/src/Internal/LoggerTest.php

```diff
@@ -0,0 +1,300 @@
+<?php
+/**
+ * Class LoggerTest
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay\Tests;
+
+use WCPAY_UnitTestCase;
+use WC_Logger;
+use WC_Log_Levels;
+use WC_Payment_Gateway_WCPay;
+use Exception;
+use WCPay\Core\Mode;
+use WCPay\Internal\Logger;
+
+/**
+ * Internal Logger tests.
+ *
+ */
+class LoggerTest extends WCPAY_UnitTestCase {
+	/**
+	 * Holds the logger, which will be tested.
+	 *
+	 * @var Logger
```

Comment:
Doing this will allow editors to understand what methods are available:

```suggestion
	 * @var Logger|MockObject
```

Doing this will remove a lot of problems:

<img width="469" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/5311119/f232e953-1673-4d85-8e9a-247ddaf2edbe">

Reply:
Done

Reply:
Almost there: You need `use PHPUnit\Framework\MockObject\MockObject;` for this to work.

Just curious, does PHPStorm understand what's going on without this definition?